### DISCUSSION TITLE
Tables/Fields/Relationships created in Database are not working

### DISCUSSION THREAD

**Discussion by ammaraga** [directus/directus#14022] [Archive]
Hi!
If I create a collection in Directus, set all the relationships... everything works fine. But I'm having issues with making the already existing MS-SQL tables work in Directus. 
I've just created a table on DB-Server, granted the permissions, but when I try to import collection, no fields are visible. Any idea what am I doing wrong?

```sql
CREATE TABLE [dbo].[pside_projectkey_ym_value](
	[ym] [date] NOT NULL,
	[projectkey_id] [int] NOT NULL,
	[value] [decimal](8, 2) NULL
) ON [PRIMARY]
GO

ALTER TABLE [dbo].[pside_projectkey_ym_value]  WITH CHECK ADD  CONSTRAINT [FK_pside_projectkey_ym_value_pside_cal_month] FOREIGN KEY([ym])
REFERENCES [dbo].[pside_cal_month] ([cal_month])
GO

ALTER TABLE [dbo].[pside_projectkey_ym_value] CHECK CONSTRAINT [FK_pside_projectkey_ym_value_pside_cal_month]
GO

ALTER TABLE [dbo].[pside_projectkey_ym_value]  WITH CHECK ADD  CONSTRAINT [FK_pside_projectkey_ym_value_pside_prgr] FOREIGN KEY([projectkey_id])
REFERENCES [dbo].[pside_prgr] ([projectkey_id])
GO

ALTER TABLE [dbo].[pside_projectkey_ym_value] CHECK CONSTRAINT [FK_pside_projectkey_ym_value_pside_prgr]
GO


GRANT select, insert, update, delete, alter ON dbo.[pside_projectkey_ym_value] TO svc_cms
GRANT select, insert, update, delete, alter ON dbo.[pside_prgr] TO svc_cms
GRANT select, insert, update, delete, alter ON dbo.[pside_cal_month] TO svc_cms
```

![image](https://user-images.githubusercontent.com/74614063/174797650-2a4dca4d-0afb-4d55-a776-2ac4d8372231.png)

**Comment by rijkvanzanten** [comment]
Your SQL doesn't create a primary key in this table, which in turn means Directus will ignore it 👍🏻

**Reply by ammaraga** [reply]
OK, got it, thank you. 
My (next) issue is, pside_cal_month has a primary key which is of type = date, and I can't seem to be able to create a relation from pside_projectkey_ym_value table that would be visible by Directus. Also, when I create a primary key in Directus, it doesn't seem to work with date type. Or am I doing something wrong?
I've also tried creating unique-keys (type=int) on server itself, in addition to primary-key=date, but Directus only sees primary keys, not other unique keys. So now I don't know how to create a relationship.

**Reply by rijkvanzanten** [reply]
Directus itself only supports auto-increment IDs, manual strings, and UUIDs as the primary key types that can be configured through Directus itself. You might be able to make it work by setting the primary key in the database directly, but I've never tried using a timestamp as a PK so your mileage may vary 👍🏻
